openbook_auth/models.py

```diff
@@ -1999,6 +2019,76 @@ def get_posts_for_user(self, user, max_id=None, min_id=None):
 
         return results
 
+    def exclude_community_from_top_posts(self, community):
+        check_can_exclude_community(user=self, community=community)
+
+        TopPostCommunityExclusion = get_top_post_community_exclusion_model()
+        top_post_community_exclusion = TopPostCommunityExclusion(
+            user=self,
+            community=community
+        )
+        self.top_posts_community_exclusions.add(top_post_community_exclusion, bulk=False)
+
+    def exclude_community_with_name_from_top_posts(self, community_name):
+        Community = get_community_model()
+        community_to_exclude = Community.objects.get(name=community_name)
+        self.exclude_community_from_top_posts(community_to_exclude)
+
+    def remove_exclusion_for_community_from_top_posts(self, community):
+        check_can_remove_exclusion_for_community(user=self, community=community)
+
+        TopPostCommunityExclusion = get_top_post_community_exclusion_model()
+        TopPostCommunityExclusion.objects.get(user=self, community=community).delete()
+
+    def remove_exclusion_for_community_with_name_from_top_posts(self, community_name):
+        Community = get_community_model()
+        community = Community.objects.get(name=community_name)
+        self.remove_exclusion_for_community_from_top_posts(community)
+
+    def get_top_posts(self, max_id=None, min_id=None, count=10):
```

Comment:
This query might make either the database blow up, or return duplicates. We need to test it on a production like db too.

Reply:
Shouldn't this also be if max_id, else if min id? And not both together? What happens if I do maxId 90, minId 98

Reply:
[1] Yea its heavily inspired by what we do on the timeline api, and the no of top posts will always be less, since its a subset of the total posts always.

[2]Yea, good catch, initially I had an else but then removed it since we were going to scroll to a position like we do in post comments, which requires min_id = max_id , but we dont do that anymore so will fix 👍